frontend/src/features/env/AdminFeedbackModal.tsx

```diff
@@ -0,0 +1,171 @@
+  const handleClose = useCallback(() => {
+    onClose()
+  }, [onClose])
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleClose}
+      size={modalSize}
+      initialFocusRef={initialRef}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalCloseButton />
+        <chakra.form noValidate onSubmit={handleSubmitForm}>
+          <ModalHeader pr="3rem">Share your feedback</ModalHeader>
+          <ModalBody mt="0" pt="0">
+            <Skeleton isLoaded={!isLoading}>
+              <Stack spacing="1rem">
+                <Input type="hidden" {...register('url')} value={url} />
+
+                <FormControl isRequired isInvalid={!isEmpty(errors)}>
+                  <FormLabel>How was your experience using FormSG?</FormLabel>
+                  <Controller
+                    rules={{ required: REQUIRED_ERROR }}
+                    control={control}
+                    name="rating"
+                    render={({ field: { value, onChange, ...rest } }) => (
+                      <Rating
+                        numberOfRatings={5}
+                        variant="star"
+                        helperText={'1: Poor, 5: Excellent'}
+                        value={Number(value)}
+                        onChange={(val) => onChange(val?.toString())}
+                        {...rest}
+                      />
+                    )}
+                  />
+                  <FormErrorMessage>
+                    {errors['rating']?.message}
+                  </FormErrorMessage>
+                </FormControl>
+
+                <FormControl>
+                  <FormLabel>Do you have any other feedback for us?</FormLabel>
+                  <Textarea {...register('feedback')} />
+                  <FormErrorMessage>
+                    {errors['feedback']?.message}
+                  </FormErrorMessage>
+                </FormControl>
+
+                {user ? (
+                  <Input
+                    type="hidden"
+                    {...register('email')}
+                    value={user.email}
+                  />
+                ) : null}
+
+                {rumSessionId ? (
+                  <Input
+                    type="hidden"
+                    {...register('rumSessionId')}
+                    value={`https://app.datadoghq.com/rum/replay/sessions/${rumSessionId}`}
+                  />
+                ) : null}
+              </Stack>
+            </Skeleton>
+          </ModalBody>
+          <ModalFooter mt={{ base: '2.5rem', md: '0' }}>
+            <Stack
+              w="100%"
+              spacing="1rem"
+              justify="right"
+              direction={{ base: 'column-reverse', md: 'row' }}
+            >
+              <Button isFullWidth={isMobile} variant="clear" onClick={onClose}>
+                Cancel
+              </Button>
+              <Button
+                isFullWidth={isMobile}
+                type="submit"
+                isDisabled={isLoading}
```

Comment:
Nit: should we just use the isLoading prop? Adds a spinner when loading too.

Reply:
I thought about it, but i decided not to use it since the submit button in a spinner state gives the impression that a submission is being made. I thought that disabling the button made more sense if the form hasn't loaded yet haha

Reply:
Ok it's not a post request? then should we disable on fetch and set loading on post then?

Reply:
yup, done in [086308e](https://github.com/opengovsg/FormSG/pull/5411/commits/086308e40d434b870badf2f25189796b2e73743e)!